Developer relations actions for new MeiliSearch releases (#76)
meilisearch/devrel: processes/releases.md

# New MeiliSearch releases - what to do

Those are the steps we need to take every time we release a new version of MeiliSearch.

## Responsibilities

> Right now, unless mentioned, the tasks for a new release on the developer relations side are divided in three categories or actions with each person accountable for them:

Reviewer: ```suggestion
Right now, unless mentioned otherwise, the tasks for a new release on the developer relations side are divided into three categories or actions with each person accountable for them:
```

- Documentation & guides: [@react-learner](https://github.com/react-learner)
- Blog posts: [@CaroFG](https://github.com/CaroFG)
- Social media: [@fharper](https://github.com/fharper)

## After the last release

1. Create a PR on a branch called prepare-meilisearch-[YYYY]-[WW] (i.e.: prepare-meilisearch-2020-03 for the third week of the year 2020) in the [documentation repository](https://github.com/meilisearch/documentation).

> 1. Ensure that all new PRs related to the next release are done on the mentioned above.

Reviewer 2: I did this by adding every PR from the MeiliSearch Milestone in the body of the prepare-meilisearch-[YYYY]-[WW] body like this: 

<img width="671" alt="Screenshot 2021-01-06 at 09 42 07" src="https://user-images.githubusercontent.com/33010418/103747791-79971300-5003-11eb-8330-293a6c8dbf0c.png">

When I overlined a PR it meant that it did not need to be documented. When it did, I added a checkbox in front that I only checked once the documentation for this PR was done, reviewed, and merged in the prepare-meilisearch-[YYYY]-[WW] branch. 
That way, if any was to be forgotten, we would know very quickly

Reviewer: This sentence is not clear for me; I understand it thanks to Charlotte's example.

Author: I updated the text and steps. Open another conversation if needed. thanks for the suggestion!

> 1. Ensure that all new PRs related to the next release are done on the mentioned above.

Reviewer: ```suggestion
2. Ensure that all new PRs related to the next release are done on the mentioned above.
```

Author: Same comment about Markdown ordered lists.

> 1. The documentation must contain at minimum the references of the new feature(s), if any.

Reviewer: ```suggestion
3. The documentation must contain at minimum the references of the new feature(s), if any.
```

Author: Same comment about Markdown ordered lists.

> 1. The decision should be made if we need to create a [guide](https://docs.meilisearch.com/guides/) for the new feature(s), if any.

Reviewer: ```suggestion
4. The decision should be made if we need to create a [guide](https://docs.meilisearch.com/guides/) for the new feature(s), if any.
```

Author: Same comment about Markdown ordered lists.

## Pre-release week

> 1. If the release has substantial new features, we should draft a [blog](https://blog.meilisearch.com/) post about the new release.

Reviewer 2: About the blog post, based a bit on what RUST does when ever they publish a new release: https://blog.rust-lang.org/2020/12/31/Rust-1.49.0.html, I try to be as explanatory as possible. 
For example, about the tokenizer, I would explain a little how it works and what we had before. In order for users to observe that we are evolving. As it took us so long to make that change, it should be at least a bit praised. 

So I would say: 
- new features (may they impact the user or not)
- new addition to the API (clearly showing with example how to use it)
- Breaking changes (for example the default value of the payload limit going from `10mb` to `100mb` is a breaking change that should be communicated).

Those are all what I use to do, it is of course up to you guys to decide if this is a direction you like. 
For the last bit of information, the release article is always popular when they are published. So I think people like to read what is new. So don't be shy :D

Author: Added those details, thanks @bidoubiwa

> 1. The prepare-meilisearch-[YYYY]-[WW] should be reviewed and ready to merge to the [master branch](https://github.com/meilisearch/documentation/tree/master).

Reviewer: ```suggestion
2. The prepare-meilisearch-[YYYY]-[WW] should be reviewed and ready to merge to the [master branch](https://github.com/meilisearch/documentation/tree/master).
```

Author: Same comment about Markdown ordered lists.

## Release day

> 1. The prepare-meilisearch-[YYYY]-[WW] branch should be merged with the [master brand](https://github.com/meilisearch/documentation/tree/master), and updated version of the documentation published [in production](https://docs.meilisearch.com/).

Reviewer: ```suggestion
1. The prepare-meilisearch-[YYYY]-[WW] branch should be merged with the [master branch](https://github.com/meilisearch/documentation/tree/master), and updated version of the documentation published [in production](https://docs.meilisearch.com/).
```

> 1. Once in production, we should publish the blog post, if any.

Reviewer: ```suggestion
2. Once in production, we should publish the blog post, if any.
```

Author: @CaroFG : maybe I'm missing something, but there is no change in this suggestion?

Author: Oh I get it, with Markdown, you can do ordered list with only 1. for each items, and the result generated will put the right numbers. I prefer to do it like that, so I don't have to change the number if I add items or reorganized them. I put specific numbers after list that were splited so the generate list continue at the right number.

_For the next steps, we will share the blog post: if none, we will share the [release notes](https://github.com/meilisearch/MeiliSearch/releases) from GitHub._

3. Share in the #general channel of our [Slack community](https://slack.meilisearch.com/) as yourself;

> 1. Share on [GitHub Discussions](https://github.com/meilisearch/MeiliSearch/discussions) as yourself;

Reviewer: ```suggestion
4. Share on [GitHub Discussions](https://github.com/meilisearch/MeiliSearch/discussions) as yourself;
```

Author: Same comment about Markdown ordered lists.

> 1. Share on [Twitter](https://twitter.com/meilisearch) (from Mention);

Reviewer: ```suggestion
5. Share on [Twitter](https://twitter.com/meilisearch) (from Mention);
```

Author: Same comment about Markdown ordered lists.

_If the release contains enough interesting new features, we should:_

6. share on [LinkedIn](https://www.linkedin.com/company/meilisearch) (need to be admin of the LinkedIn page);

> 1. share on [Reddit](https://www.reddit.com/) (need to add the sub/subs) (credentials in 1Password);

Reviewer: ```suggestion
7. share on [Reddit](https://www.reddit.com/) (need to add the sub/subs) (credentials in 1Password);
```

Author: Same comment about Markdown ordered lists.
